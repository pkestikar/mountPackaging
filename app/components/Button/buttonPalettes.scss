.mp--button-primary {
  background: $navy-blue;
  color: white;
  border: none;
  box-shadow: 0 2px 8px 0 $blue-grey;
}
.mp--button-primary:hover {
  box-shadow: 0 2px 8px 0 white;
}
.mp--button-primary:active {
  box-shadow: 0 2px 8px 0 white;
}

.mp--button-secondary {
  background: white;
  color: $navy-blue;
  border: 1px solid $navy-blue;
  box-shadow: 0 2px 8px 0 rgba(127, 133, 149, 0.28);
}
.mp--button-secondary:hover {
  box-shadow: 0 2px 8px 0 $navy-blue;
}
.mp--button-secondary:active {
  box-shadow: 0 2px 8px 0 $navy-blue;
}

.mp--button-tertiary {
  background: $dark-blue;
  color: white;
  border: 1px solid $dark-blue;
  box-shadow: 0 2px 8px 0 rgba(127, 133, 149, 0.28);
}
.mp--button-tertiary:hover {
  box-shadow: 0 2px 8px 0 $navy-blue;
}
.mp--button-tertiary:active {
  box-shadow: 0 2px 8px 0 $navy-blue;
}
