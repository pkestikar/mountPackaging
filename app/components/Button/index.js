import React from 'react';
import classnames from 'classnames';

const Button = (props) => {
    const componentName = 'mp--button';
    const classes = classnames(
        `${componentName}`,
        {[`${props.customClass}`]: !!props.customClass},
        {[`loading`]: !!props.isLoading},
        {['disabled']: !!props.disabled},
        {[`small`]: props.size === 'small'},
        {[`medium`]: props.size === 'medium'},
        {[`large`]: props.size === 'large'},
        {[`${componentName}-${props.palette}`]: !!props.palette}
    );

    return (
        <button className={classes} onClick={() => props.onClick()}>
            {props.isLoading ?
                'Loading..'
                :
                props.children
            }
        </button>
    )
};

Button.defaultProps = {
    size: 'medium',
    disabled: false,
    loading: false,
    palette: 'primary'
};

export default Button;
