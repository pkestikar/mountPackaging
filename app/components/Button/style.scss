@import './buttonPalettes';

.mp--button {
  height: 45px;
  width: 175px;
  &.small {
    height: 35px;
    width: 145px;
  }
  &.medium {
    height: 45px;
    width: 175px;
  }
  &.large {
    height: 55px;
    width: 200px;
  }

  display: flex;
  justify-content: center;
  align-items: center;
  outline: none;
  transition: all 0.2s ease;

  &.disabled {
    //pointer-events: none;
    cursor: not-allowed;
  }
}

