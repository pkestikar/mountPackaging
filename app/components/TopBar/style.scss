.mp--topbar {
  display: flex;
  z-index: 9999;
  justify-content: center;
  width: 100%;
  position: fixed;
  top: 0;
  left: 0;
  height: $top-bar-offset-height__large;
  background: transparent;
  transition: all 0.3s ease;
  &.scroll {
    background: white;
  }

  &--inner {
    //max-width: $site-max-width;
    height: 100%;
    //display: flex;
  }

  .topbar-grid-container {
    height: 100%;
  }

  .logo-container {
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    float: left;
    @media only screen and (max-width: $medium) {
      img {
        margin: 1rem;
      }
    }
  }

  .bm-item {
    color: $light-periwinkle;
    text-decoration: none;
    transition: all 0.2s ease;
    outline: none !important;
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 1rem 0;
  }
  .bm-item:hover {
    color: $light-periwinkle;
    text-decoration: underline;
  }
  .bm-item:active {
    color: $light-periwinkle;
    text-decoration: underline;
  }

  .bm-menu-wrap {
    background: $navy-blue;
  }

  .topbar-elements-container {
    height: 100%;
    display: flex;
    justify-content: flex-start;
    align-items: center;
    a {
      width: 7rem;
      text-align: center;
      text-decoration: none;
      color: $neutral;
    }
    @media only screen and (max-width: $medium) {
      a {
        width: 6rem;
      }
    }
    a:hover {
      text-decoration: underline;
      color: $dark-blue;
    }
    a:active {
      text-decoration: underline;
      color: $dark-blue;
    }

    .active {
      text-decoration: underline;
      color: $dark-blue;
    }
  }

  .get-quote-wrapper {
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    margin-left: auto;
    //button {
    //  margin: 3rem;
    //}
    @media only screen and (max-width: $medium) {
      button {
        margin: 1rem;
      }
    }
  }
}