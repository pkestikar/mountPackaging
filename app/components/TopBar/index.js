import React from 'react';
import { NavLink } from "react-router-dom";
import Button from 'Components/Button';

const TopBar = (props) => {
    return (
        <div className="mp--topbar">
            <div className="logo-container">
                <img src="../../../images/logo.png" alt="Logo"/>
            </div>
            <div className="topbar-elements-container">
                <NavLink to={'/about-us/'} activeClassName={'active'}>About Us</NavLink>
                <NavLink to={'/work/'} activeClassName={'active'}>Work</NavLink>
                <NavLink to={'/services/'} activeClassName={'active'}>Services</NavLink>
                <NavLink to={'/contact/'} activeClassName={'active'}>Contact</NavLink>
            </div>
            <div className="get-quote-wrapper">
                <Button palette={'secondary'} isLoading={true}>Get Quotation</Button>
            </div>
        </div>
    )
};

export default TopBar;

