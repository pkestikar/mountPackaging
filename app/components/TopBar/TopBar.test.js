import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import TopBar from './index';
import Button from '../Button';

// jest.mock('../Button', () => {
//     return {
//         __esModule: true,
//         default: () => {
//             return (
//                 <button>
//                     Button Text
//                 </button>
//             )
//         }
//     }
// });

jest.mock('../Button', () => 'Button');

describe('TopBar Component', () => {

    test('if it renders correctly', () => {
        const wrapper = shallow(<TopBar />);

        expect(toJSON(wrapper)).toMatchSnapshot();
    });

});
