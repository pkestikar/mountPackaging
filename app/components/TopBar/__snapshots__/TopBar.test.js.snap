// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar Component if it renders correctly 1`] = `
<div
  className="mp--topbar"
>
  <div
    className="logo-container"
  >
    <img
      alt="Logo"
      src="../../../images/logo.png"
    />
  </div>
  <div
    className="topbar-elements-container"
  >
    <NavLink
      activeClassName="active"
      to="/about-us/"
    >
      About Us
    </NavLink>
    <NavLink
      activeClassName="active"
      to="/work/"
    >
      Work
    </NavLink>
    <NavLink
      activeClassName="active"
      to="/services/"
    >
      Services
    </NavLink>
    <NavLink
      activeClassName="active"
      to="/contact/"
    >
      Contact
    </NavLink>
  </div>
  <div
    className="get-quote-wrapper"
  >
    <Button
      palette="secondary"
    >
      Get Quotation
    </Button>
  </div>
</div>
`;
